test/mainnet/curve.spec.sol

```diff
@@ -0,0 +1,229 @@
+                isActive: true,
+                paramRules: exchangeRules,
+                validator: IValidator(address(0))
+            }));
+        }
+        // Set approval rule to allow ethStethPool to spend stETH tokens from the vault
+        setup.setApprovalRule(_vault, MC.STETH, ethStethPool);
+
+        // Set approval rules for ynETH and wstETH to be spent by ynETHWstETH pool
+        setup.setApprovalRule(_vault, MC.YNETH, ynETHWstETHPool);
+        setup.setApprovalRule(_vault, MC.WSTETH, ynETHWstETHPool);
+
+        vm.stopPrank();
+    }
+
+    function test_Vault_Curve_swapStETHtoETH() public {
+        uint256 amount = 100 ether;
+
+        // Create user and give them ETH
+        address user = makeAddr("user");
+        deal(user, 100000 ether);
+
+        // Convert ETH to stETH via Lido
+        vm.startPrank(user);
+        IStETH(MC.STETH).submit{value: amount + 100 wei }(address(0));
+        vm.stopPrank();
+
+        // Approve and deposit stETH to vault
+        vm.startPrank(user);
+        IERC20(MC.STETH).approve(address(vault), amount);
+        vault.depositAsset(MC.STETH, amount, user);
+        vm.stopPrank();
+
+        // Read total assets before swap
+        uint256 totalAssetsBefore = vault.totalAssets();
+
+
+        // Exchange stETH for ETH via Curve pool
+        ICurveRegistry registry = ICurveRegistry(MC.CURVE_REGISTRY);
+        ICurvePool pool = ICurvePool(registry.find_pool_for_coins(MC.ETH, MC.STETH));
+
+        uint256 swapAmount = amount / 2;
+
+        uint256 minOut = pool.get_dy(1, 0, swapAmount); // Swap from stETH (1) to ETH (0)
+
+        uint256 delta = swapAmount - minOut + 2 wei; // wei difference from stETH balance error
+
+        // Prepare approve data
+        bytes memory approveData = abi.encodeWithSelector(
+            IERC20.approve.selector,
+            address(pool),
+            swapAmount
+        );
+
+        // Prepare exchange data
+        bytes memory exchangeData = abi.encodeWithSelector(
+            bytes4(keccak256("exchange(int128,int128,uint256,uint256)")),
+            1, // stETH index
+            0, // ETH index
+            swapAmount,
+            minOut
+        );
+
+        {
+            address[] memory targets = new address[](2);
+            targets[0] = MC.STETH;
+            targets[1] = address(pool);
+
+            uint256[] memory values = new uint256[](2);
+            values[0] = 0;
+            values[1] = 0;
+
+            bytes[] memory callData = new bytes[](2);
+            callData[0] = approveData;
+            callData[1] = exchangeData;
+
+            vm.startPrank(PROCESSOR);
+            vault.processor(targets, values, callData);
+            vm.stopPrank();
+        }
+
+        // Assert stETH balance is 0 and ETH balance matches expected amount
+        assertApproxEqAbs(IERC20(MC.STETH).balanceOf(address(vault)), amount - swapAmount, 2);
+        assertApproxEqAbs(address(vault).balance, minOut, 2);
```

Comment:
seems this eth is stuck in the vault?

Reply:
you can process it with a call to function processor(), wrap it as WETH, deposit it somewhere else etc.